### PR TITLE
DEVDOCS 3716

### PR DESCRIPTION
**Pull request by bc-tgomez**
# [DEVDOCS-3716](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3716)

## Create REFUND_ID endpoint
* Add REFUND ID endpoint to Orders V3

### CODE REVIEW

**Comment by bc-tgomez** [comment]
@bigcommerce/orders Could I get a review please? 🐸

File: reference/orders.v3.yml [hunk 1]
```diff
@@ -416,6 +416,50 @@ paths:
...
+            type: string
+        - $ref: '#/components/parameters/Accept'
+        - $ref: '#/components/parameters/Content-Type'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RefundID_Get'
+        required: true
+        x-examples: null
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:448]: we should not have "requestBody" for GET requests

File: reference/orders.v3.yml [hunk 2]
```diff
@@ -416,6 +416,50 @@ paths:
...
+      responses:
+        '201':
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:450]:
```suggestion
        '200':
```

File: reference/orders.v3.yml [hunk 3]
```diff
@@ -416,6 +416,50 @@ paths:
...
+          description: Return a refund with 'Refund_ID'.
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:451]:
```suggestion
          description: Return a refund by refund ID.
```

I think this line can be removed

File: reference/orders.v3.yml [hunk 4]
```diff
@@ -416,6 +416,50 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/RefundID_Get'
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:460]: `data` is an object, not an array.

Example response:

image

File: reference/orders.v3.yml [hunk 5]
```diff
@@ -2374,6 +2418,136 @@ components:
         - items
         - payments
       x-examples: {}
+    RefundID_Get:
+      type: object
+      properties:
+        data:
+          type: object
+          properties:
+            refund_id:
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2427]:
```suggestion
            id:
```

File: reference/orders.v3.yml [hunk 6]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+              description: |
+                Timestamp of when the refund was created
+            reason:
+              type: string
+              description: |
+                Reason for refund
+            total_amount:
+              type: number
+              description: |
+               A non-negative 2 decimal place rounded value that represents the amount that can be charged/refunded via payment providers.
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2450]: is `charged/refunded` correct? or should it be only `refunded` ? @bc-vincent-zhao

File: reference/orders.v3.yml [hunk 7]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                    enum:
+                    - PRODUCT
+                    - GIFT_WRAPPING
+                    - SHIPPING
+                    - HANDLING
+                    - ORDER
+                  item_id:
+                    type: integer
+                    description: |
+                     order_product.id corresponding to the item_types of PRODUCT, GIFT_WRAPPING. order_address.id corresponding to the item_types of SHIPPING, HANDLING. order.id corresponding to the item_type of ORDER.
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2509]:
```suggestion
                     `order_product.id` corresponding to the item_types of PRODUCT, GIFT_WRAPPING. `order_address.id` corresponding to the item_types of SHIPPING, HANDLING. `order.id` corresponding to the item_type of ORDER.
```

File: reference/orders.v3.yml [hunk 8]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                  quantity:
+                    type: integer
+                    description: |
+                     Quantity of item refunded. Note: this will only be populated for item_type PRODUCT
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2513]: this field is `nullable`

File: reference/orders.v3.yml [hunk 9]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                  requested_amount:
+                    type: 'string'
+                    description: |
+                     A non-negative 2 decimal place rounded value that represents the amount that can be charged/refunded via payment providers
+                    example: '109.11'
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2518]: this field is `nullable`

File: reference/orders.v3.yml [hunk 10]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                    type: string
+                    description: |
+                     Reason for refunding an item
+        meta:
+          type: object
+          properties: {}
+      x-examples:
+        example-1:
+          data:
+            refund_id: 1
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2529]:
```suggestion
            id: 1
```

File: reference/orders.v3.yml [hunk 11]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                amount: 109.11
+                offline: true
+                is_declined: false
+                declined_message: ''
+            items:
+              - item_type: PRODUCT
+                item_id: 17
+                quantity: 1
+                requested_amount: 109.11
+                reason: string
```
Comment (donald-nguyen-bc) [inline reference/orders.v3.yml:2549]:
```suggestion
                reason: ''
```

**Comment by donald-nguyen-bc** [comment]
Thank you @bc-tgomez, overall looks good, just some minor comments

File: reference/orders.v3.yml [hunk 12]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+               Whether refund amount and tax are provided explicitly by merchant override.
+            payments:
+              type: array
+              items:
+                type: object
+                properties:
+                  id:
+                    type: integer
+                    description: |
+                     Reference to refumd payment ID
```
Comment (traci707) [inline reference/orders.v3.yml:2468]:
```suggestion
                     Reference to refund payment ID
```

File: reference/orders.v3.yml [hunk 13]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+              description: |
+                Refund_ID for the return
+            order_id:
+              type: integer
+              description: |
+               Reference to order id
+            user_id:
+              type: integer
+              description: |
+                Reference to the user'd ID who created the refund. This is automatically populated by BigCommerce.
```
Comment (traci707) [inline reference/orders.v3.yml:2438]:
```suggestion
                Reference to the user ID who created the refund. This is automatically populated by BigCommerce.
```

File: reference/orders.v3.yml [hunk 14]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                     Reference to refumd payment ID
+                  provider_id:
+                    type: string
+                    description: |
+                     Reference to payment provider
+                     example: storecredit
+                  amount:
+                    type: number
+                    description: |
+                     A non-negative 2 decimal place rounded value that represents the amount that can be charged/refunded via payment providers.
```
Comment (traci707) [inline reference/orders.v3.yml:2477]:
```suggestion
                     A non-negative two decimal place rounded value represents the amount that can be charged/refunded via payment providers.
```

File: reference/orders.v3.yml [hunk 15]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                    example: 109.11
+                  offline:
+                    type: boolean
+                    description: |
+                     Indicates whether payment was offline
```
Comment (traci707) [inline reference/orders.v3.yml:2482]:
```suggestion
                     Indicates whether the payment was offline.
```

File: reference/orders.v3.yml [hunk 16]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                  is_declined:
+                    type: boolean
+                    description: |
+                     Indicates if this payment has been declined by the payment provider
+                  declined_message:
+                    type: string
+                    description: |
+                     Message indicates why payment was declined
```
Comment (traci707) [inline reference/orders.v3.yml:2490]:
```suggestion
                     Message indicates why the payment was declined.
```

File: reference/orders.v3.yml [hunk 17]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                    description: |
+                     order_product.id corresponding to the item_types of PRODUCT, GIFT_WRAPPING. order_address.id corresponding to the item_types of SHIPPING, HANDLING. order.id corresponding to the item_type of ORDER.
+                  quantity:
+                    type: integer
+                    description: |
+                     Quantity of item refunded. Note: this will only be populated for item_type PRODUCT
+                  requested_amount:
+                    type: 'string'
+                    description: |
+                     A non-negative 2 decimal place rounded value that represents the amount that can be charged/refunded via payment providers
```
Comment (traci707) [inline reference/orders.v3.yml:2517]:
```suggestion
                     A non-negative two decimal place rounded value represents the amount that can be charged/refunded via payment providers
```

File: reference/orders.v3.yml [hunk 18]
```diff
@@ -416,6 +416,50 @@ paths:
         required: true
         schema:
           type: string
+  '/stores/{store_hash}/v3/orders/payment_actions/refunds/{refund_id}':
+    get:
+      summary: Return a refund by refund ID.
```
Comment (traci707) [inline reference/orders.v3.yml:421]:
```suggestion
      summary: Returns a refund by refund ID.
```

File: reference/orders.v3.yml (hunk 13 above)
Comment (traci707) [inline reference/orders.v3.yml:2438]:
```suggestion
                Reference to the user ID who created the refund. This is automatically populated by BigCommerce.
```

File: reference/orders.v3.yml (hunk 6 above)
Comment (traci707) [inline reference/orders.v3.yml:2450]:
```suggestion
               A non-negative two decimal place rounded value represents the amount that can be charged/refunded via payment providers.
```

File: reference/orders.v3.yml [hunk 19]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                     A non-negative 2 decimal place rounded value that represents the amount that can be charged/refunded via payment providers.
+                    example: 109.11
+                  offline:
+                    type: boolean
+                    description: |
+                     Indicates whether payment was offline
+                  is_declined:
+                    type: boolean
+                    description: |
+                     Indicates if this payment has been declined by the payment provider
```
Comment (traci707) [inline reference/orders.v3.yml:2486]:
```suggestion
                     Indicates if this payment has been declined by the payment provider.
```

File: reference/orders.v3.yml [hunk 20]
```diff
@@ -2374,6 +2418,136 @@ components:
...
+                     Message indicates why payment was declined
+            items:
+              type: array
+              items:
+                type: object
+                properties:
+                  item_type:
+                    type: string
+                    description: |
+                     Type of item that was refunded
```
Comment (traci707) [inline reference/orders.v3.yml:2499]:
```suggestion
                     Type of item that was refunded.
```

**Review by traci707 — COMMENTED** [review]
Added some editorial suggestions.

**Review by markcmurphy — APPROVED** [review]
Made a change to the summary so it was consistent with other titles in the left nav + a few more minor changes

**Pull request merged**